src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        threshold = (int) (LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V>[] tab = table;
+        int elementIndex;
+        if (tab == null || tab.length == 0) {
+            resize();
+        }
+        elementIndex = Math.abs(hash(key) % tab.length);
+        if (tab[elementIndex] == null) {
+            tab[elementIndex] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        Node<K, V> head = tab[elementIndex];
+        while (head != null) {
+            if (Objects.equals(head.key, key)) {
+                head.value = value;
+                break;
+            }
+            if (head.next == null) {
+                head.next = new Node<>(key, value, null);
+                size++;
+                break;
+            }
+            head = head.next;
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> head;
+        Node<K, V> returnValue = null;
+        Node<K, V>[] tab = table;
+        head = tab[Math.abs(hash(key) % tab.length)];
+        while (head != null) {
+            if (Objects.equals(key, head.key)) {
+                return head.value;
+            }
+            head = head.next;
+        }
+        if (returnValue != null) {
+            return returnValue.value;
+        }
```

Comment:
do we need this if block?

Reply:
no, we needn't, like returnValue variable